service/tracker/execution.go

```diff
@@ -159,7 +171,40 @@ func (e *Execution) processNext() error {
 	// Dump the block execution record into our cache and push all trie updates
 	// into our update queue.
 	e.records[blockID] = record
```

Comment:
should we dump to cache after the validation instead of before

Reply:
validation throws an error below and stops the process, so doesn't matter where we do this.